crypto/src/main/kotlin/web5/sdk/crypto/Ed25519.kt

```diff
@@ -156,8 +190,8 @@ public object Ed25519 : KeyGenerator, Signer {
    * @param key The [JWK] to validate.
    * @throws IllegalArgumentException if the key is not of type [OctetKeyPair] or if the key type is not [KeyType.EC].
    */
-  public fun validateKey(key: JWK) {
-    require(key is OctetKeyPair) { "private key must be an Octet Key Pair (kty: OKP)" }
+  private fun validateKey(key: JWK) {
+    require(key is OctetKeyPair) { "key must be an Octet Key Pair (kty: OKP)" }
```

Comment:
I'm not as familiar with what is being done with these checks but are these two lines redundantly checking for the same thing? (i.e., that `kty` is OKP?)

Or does `key is OctetKeyPair` go beyond just checking the `kty` property?

```kotlin
    require(key is OctetKeyPair) { "key must be an Octet Key Pair (kty: OKP)" }
    require(key.keyType == keyType) { "private key key type must be OKP" }
```

Reply:
I think the first one is a runtime type check. `OctetKeyPair` extends `JWK` so it's possible that `key` does not conform to OKP type. 

Not sure if we need the second check if the first check is true though.